### PR TITLE
Foodening

### PR DESCRIPTION
**Pull request by MTandi**
Reopen of #76205

## Changelog

:cl:
add: Mood buff power and duration depends on the ingredient quality and recipe complexity of the food.
add: Hand-crafted food also gives a non-mood buff. The buff type depends on recipe complexity and duration - on ingredient quality.
code: Added infrastructure to assign custom buff to certain dishes for unique effects
add: Consumable reagents now care about purity, being more nutritious with higher purity. Dispensers give synthetic food, natural sources are organic and better and give longer buffs.
add: Grown food reagent purity now depends on lifespan, endurance and instability stats of the seed. 
add: Natural booze now uses the same purity formula and 100% purity can increase booze power by up to 200%.
add: Added Fat reagent that is now a base type for lard, vegetable oil, olive oil. Corn oil and cooking oil merged into vegetable oil.
balance: Reagents and their purities are preserved during food chemical reactions, processing, cooking and crafting.
balance: Sugar nutrition rebalanced due to it being 100% pure in chem dispenser, overdose happens at 100u now.
fix: `grind_results` of grown food give full amount equal to all removed nutrients instead of just `/datum/reagent/consumable/nutriment`
qol: Seed Extractor now shows additional seed data: possible mutations, juicing result, grinding result, fermentation result
fix: Reverts the unintended change of monkey cube biomass cost back to 50
/:cl:

### CODE REVIEW

**Comment by YehnBeep** [comment]
Is the sugar OD reduction particularly important to this? It's going to make it a lot easier to coma people with honey bees and such.

**Comment by Mothblocks** [comment]
you have way way way too much shit in here man

**Review by Mothblocks — CHANGES_REQUESTED** [review]
please convince me why the 1500 lines of code for what looks like 10+ changelog entries needs to be one PR. your original PR got only a smidge of conversation from maintainers after weeks and that is not a good sign to me

**Comment by MTandi** [comment]
> please convince me why the 1500 lines of code for what looks like 10+

We have 800 recipes, all of which got a complexity variable assigned.

**Comment by MTandi** [comment]
> Is the sugar OD reduction particularly important to this? It's going to make it a lot easier to coma people with honey bees and such.

Not really. It's just that current sugar pills don't have any downside over other food, even if you eat it to fill from zero to full satiety. Thought that reducing the threshold would make more fun when people that eat nothing but sugar during the shift have the overdose triggered.

If there is any concern that it will cause bad gameplay impact, then let me know.

**Comment by YehnBeep** [comment]
> If there is any concern that it will cause bad gameplay impact, then let me know.

Mostly just the aforementioned honey bees and being coma-locked easily but it might be too minor a thing to really matter.

**Comment by Mothblocks** [comment]
> > please convince me why the 1500 lines of code for what looks like 10+
> 
> We have 800 recipes, all of which got a complexity variable assigned.

you say this but your changelog is enormous which is what makes me skeptical that this is purely a code churn thing

**Comment by MTandi** [comment]
> > > please convince me why the 1500 lines of code for what looks like 10+
> > 
> > 
> > We have 800 recipes, all of which got a complexity variable assigned.
> 
> you say this but your changelog is enormous which is what makes me skeptical that this is purely a code churn thing

Also cooking oil, quality oil and corn oil typepath change.

**Comment by Likteer** [comment]
> please convince me

@Mothblocks Right now, all these cuisine expansions we've had seem silly.

Deep fried organs, stolen shoes and slices of watermelons 
leftover from biomass production, dominate today's spessmen food pyramid.

When MTandi's arduous work bears fruit, I am positive I will live to see the day where I encounter a ZAGOSK SURF 'N' TURF SMORGASBORD in the game  🙏

**Comment by PotatoOfChaos** [comment]
What a shame, I'd hate to see chef gameplay boil down to "make the meta pick food of the day", the fun of all the chef cuisine updates is making line-ups and menus tailored to a gimmick you come up with. :(
And the PR itself seems pretty gigantic, some of this stuff could probably be atomized.

File: code/datums/components/food/edible.dm [hunk 1]
```diff
@@ -465,6 +499,24 @@ Behavior that's still missing from this component that original food items had t
 		return
 	return TRUE
 
+///Applies food buffs according to the crafting complexity
+/datum/component/edible/proc/ApplyBuff(mob/eater)
```
Comment (jlsnow301) [inline code/datums/components/food/edible.dm:503]: Can you snake case this pls?

I know it's trying to be consistent but they're also wrong

**Comment by Likteer** [comment]
> please convince me

@Mothblocks Cooking complex dishes right now is high effort with no reward.

Instead of matching dishes to diets, people default to the cheapest available options: organs, watermelons, plain steaks, shoes.

Used to be, you'd see most of the cool dishes coming out of xenobiology. 
Then these were changed to be toxic.
I don't think it was a bad change, 
but now you see them where mappers put them.

Biogen is so convenient, it made me wonder who'd even bother getting it's boons elsewhere.
So MTandi decided to reward you for bothering.

Purity and booze power additions are a cherry on top, for knowledge from other jobs/cooperating with them.
It's the 1 thing I could see be split into another PR.

### TL;DR I think effort should be rewarded, 
and I want to live to see the day someone makes a ZAGOSK SURF 'N' TURF SMORGASBORD.

**Comment by MTandi** [comment]
> What a shame, I'd hate to see chef gameplay boil down to "make the meta pick food of the day", the fun of all the chef cuisine updates is making line-ups and menus tailored to a gimmick you come up with. :( And the PR itself seems pretty gigantic, some of this stuff could probably be atomized.

Where did you guys even take this? In this PR the food is split only by complexity. There is no meta here. You can just make a simple food and get a small movespeed buff or make a more complex food from any category to get a stronger movespeed buff. The insulation is there just as a placeholder for other buffs people come up with and assign to different complexities or specific dishes, and is useless if you have insuls, grey bull or just don't plan to hack anything.

**Comment by Quink-Fix** [comment]
I've been wanting to see more depth/complexity from cooking beyond 'here's stuff that is ridiculously hard to make and does nothing extra except look cool', and this looks to be a good start with the added bonus of encouraging botany even more to work together more with the chef/bartender.

**Comment by jlsnow301** [comment]
Still waiting on this to get test merged. Initially I like this, I think it adds some much needed depth to cooking

**Comment by Mothblocks** [comment]
It's DNM'd because the PR was too big to get meaningful review last time and it is still looks too big

**Comment by Mothblocks** [comment]
Ok I will remove the DNM because scrolling through suggests that there's not as much code as I remember there being. My initial problem was that I was able to scroll through and intermixed with recipe changes was meaningful code changes. I'm just getting really frustrated at how many several thousand diff PRs we have that don't get meaningful review
